driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -237,6 +238,7 @@ private Metrics metrics() {
      */
     class SpeculativeExecution implements Connection.ResponseCallback {
         final String id;
+        private final Message.Request request;
```

Comment:
Why store this as a final field if `callback.request()` is accessible from the enclosing `RequestHandler` instance?